### PR TITLE
remove dependent indexes on user/case deletion

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?178242

not sure who is best to review so defaulting to @emord buddy

### CODE REVIEW

File: corehq/apps/users/tests/retire.py [hunk 1]
```diff
@@ -30,6 +31,42 @@ def tearDown(self):
...
+                'case_type': 'a-case',
+                'create': True,
+            },
+        )
+        # create a parent/child set of cases
+        parent_id, child_id = [uuid.uuid4().hex for i in range(2)]
+        child, parent = factory.create_or_update_case(CaseStructure(
+            case_id=child_id,
+            relationships=[
+                 CaseRelationship(CaseStructure(case_id=parent_id))
```
Comment (dimagimon) [inline corehq/apps/users/tests/retire.py]: E126 continuation line over-indented for hanging indent

**Comment by czue** [comment]
oops just kidding. @benrudolph buddy.

File: corehq/apps/users/tasks.py [hunk 2]
```diff
@@ -37,11 +42,12 @@ def bulk_upload_async(domain, user_specs, group_specs, location_specs):
...
 
 @task(rate_limit=2, queue='background_queue', ignore_result=True)  # limit this to two bulk saves a second so cloudant has time to reindex
-def tag_docs_as_deleted(cls, docs, deletion_id):
+def tag_cases_as_deleted_and_remove_indices(domain, docs, deletion_id):
     for doc in docs:
         doc['doc_type'] += DELETED_SUFFIX
         doc['-deletion_id'] = deletion_id
-    cls.get_db().bulk_save(docs)
+    CommCareCase.get_db().bulk_save(docs)
+    _remove_indices_from_deleted_cases_task.delay(domain, [doc['_id'] for doc in docs])
```
Comment (benrudolph) [inline corehq/apps/users/tasks.py]: just double checking that you want to start another async task from within an async task

**Comment by benrudolph** [comment]
:+1: from me pending tests

File: corehq/apps/users/tasks.py (hunk 2 above)
Comment (czue) [inline corehq/apps/users/tasks.py]: yeah - i expect that we'll want to retry the second part so i'm just preparing for that

Comment (benrudolph) [inline corehq/apps/users/tasks.py]: sounds good

**Pull request merged**